backend/lib/server_code/api/roomRoutes.js

```diff
@@ -14,24 +19,40 @@ roomRouter.use((req, res, next) => {
   next();
 });
 
-roomRouter.post("/create", (req, res) => {
+roomRouter.post("/create", async (req, res) => {
   const { creatorID } = req.body;
-  //roomPlayers.push(creatorID);
+  const roomID = roomIDCreator();
   roomCreated = true;
-  res.send({ roomID: 1, creatorID, creationDate: res.date });
+  let [socket, socketPort] = [null, null];
+  try {
+    [socket, socketPort] = await createNewSocket();
+    console.log(`socket port got from function:${socketPort}`);
+    socketsCollection[roomID] = [socket, socketPort];
+    res.send({
+      roomID: roomID,
+      creatorID,
+      creationDate: res.date,
+      socketPort: socketPort,
+    });
+  } catch (e) {
+    console.log(`An error occured while getting socket: ${e}`);
+  }
+
+  console.log("here!");
 });
 
 roomRouter.get("/join", (req, res) => {
   const { userID, roomID, playerName } = req.query;
   const playerToAdd = { id: userID, name: playerName };
-  console.log("roomID:", roomID);
+  console.log(`roomID:${roomID}`);
   if (Number(roomID) === 1 && roomCreated) {
     roomPlayers.push(playerToAdd);
-    socket.emit("NewPlayer", {
-      message: "new player has joing the room",
-      roomPlayers
+    const [roomSocket, roomSocketPort] = socketsCollection[roomID];
+    roomSocket.emit("NewPlayer", {
+      message: "A new user has joined the room",
+      roomPlayers,
     });
-    res.send({ joinDate: res.date, roomPlayers });
+    res.send({ joinDate: res.date, roomPlayers, roomSocketPort });
```

Comment:
can't we decide on the port and have it as a const in the backend and frontend? or this is the best practice?

Reply:
best practice, you dont want same port because the channel for broadcasting should be for each room separately